doc/manual/src/glossary.md

```diff
@@ -115,14 +115,14 @@
     then be built.
 
   - [reference]{#gloss-reference}\
-    A store path `P` is said to have a reference to a store path `Q` if
-    the store object at `P` contains the path `Q` somewhere. The
-    *references* of a store path are the set of store paths to which it
-    has a reference.
+    A [store object] `O` is said to have a *reference* to a store object `P` if the [store path] of `P` appears in the contents of `O`.
+    The *references* of a store object `O` are the set of store objects to which `O` has a reference.
 
-    A derivation can reference other derivations and sources (but not
-    output paths), whereas an output path only references other output
-    paths.
+    Store objects can refer to both other store objects and themselves.
```

Comment:
Can a store object refer to itself if a store object doesn't know its own store path?

Reply:
Yes.

- Some of the time does know its own store path, so there is no issue. For example, when the store object is mounted in the file system

- Some of the time it does not know its own store path, so self-references are made with a sentinel value. For example, when the store object is to be content-addressed and is being hashed.

Reply:
We should add these specifics to the relevant architecture documentation section.